src/stores/BlockchainStore.ts

```diff
@@ -224,7 +224,7 @@ export default class BlockchainStore {
             chainId
           );
 
-          const fromBlock = lastCheckedBlockNumber;
+          const fromBlock = lastCheckedBlockNumber + 1;

```

Comment:
I actually wonder if this alone might fix it, do you still see duplication logs after this change?